### ISSUE TITLE
Segfault in dupClientReplyValue

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#177]
```
From: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=570371

Redis segfaults in addReply function when I run redis-benchmark against 
it and connect two slaves simultaneously. The cause seems to be a typo 
in the dupClientReplyValue function:

--- redis-1.2.1-orig/redis.c    2010-02-18 12:40:23.000000000 +0100
+++ redis-1.2.1/redis.c 2010-02-18 12:42:03.000000000 +0100
@@ -1975,7 +1975,7 @@

 static void *dupClientReplyValue(void *o) {
     incrRefCount((robj*)o);
-    return 0;
+    return o;
 }

It returned a NULL pointer that was used afterwards as a list*. The 
patched version has been running for 16+ hours now with slaves and 
redis-benchmark reconnecting periodically.
```

Original issue reported on code.google.com by `ChrisLam...@gmail.com` on 9 Mar 2010 at 4:25

**Comment by GoogleCodeExporter** [comment]
```
Thanks, that's already fixed in Redis Git. Don't remember if it's ok in Redis 
1.2.4 as well 
but I'll backport the fix ASAP.
```

Original comment by `anti...@gmail.com` on 9 Mar 2010 at 5:07

**Comment by GoogleCodeExporter** [comment]
```
It isn't fixed in 1.2.4  (just went to release that and noticed the patch still 
applied)
```

Original comment by `ChrisLam...@gmail.com` on 9 Mar 2010 at 5:12

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 1.2.5, thanks :)
```

Original comment by `ChrisLam...@gmail.com` on 11 Mar 2010 at 9:37

**Comment by GoogleCodeExporter** [comment]
Original comment by `anti...@gmail.com` on 24 Aug 2010 at 3:18
- Changed state: **Verified**
